### PR TITLE
Update ImportError items instead of deleting and recreating them

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Each time a dag with import error is parsed, the ImportError record is deleted
and a new record is created. For example, say I have two dags with import errors,
initially, the import error id will be dag_1:import_error.id=1, dag2:import_error.id=2.
In the next dag parsing, the import error will increase. dag_1:import_error.id=3,
dag_2:import_error.id=4 and it continues like that.
This makes it impossible for the get import error REST API endpoint to be consistent

This PR fixes this issue by updating the existing record and creating a new one if no record
exists

### CODE REVIEW

File: airflow/dag_processing/processor.py [hunk 1]
```diff
@@ -528,23 +529,34 @@ def update_import_errors(session: Session, dagbag: DagBag) -> None:
         :param session: session for ORM operations
         :param dagbag: DagBag containing DAGs with import errors
         """
-        # Clear the errors of the processed files
-        for dagbag_file in dagbag.file_last_changed:
-            session.query(errors.ImportError).filter(
-                errors.ImportError.filename.startswith(dagbag_file)
-            ).delete(synchronize_session="fetch")
+        last_changed = copy.copy(dagbag.file_last_changed)
```
Comment (jedcunningham) [inline airflow/dag_processing/processor.py]: I can't say I like this variable name. I had to walk this whole section to figure out what it was.

Maybe instead we name this `all_parsed_files`, keep a separate set of files that had errors, and iterate `all_parsed_files - files_with_errors`?

File: airflow/dag_processing/processor.py [hunk 2]
```diff
@@ -528,23 +529,34 @@ def update_import_errors(session: Session, dagbag: DagBag) -> None:
...
+        import_error_files = [x.filename for x in session.query(errors.ImportError.filename).all()]
```
Comment (jedcunningham) [inline airflow/dag_processing/processor.py]: Maybe `existing_import_error_files`?

File: airflow/dag_processing/processor.py [hunk 3]
```diff
@@ -528,23 +529,36 @@ def update_import_errors(session: Session, dagbag: DagBag) -> None:
         :param session: session for ORM operations
         :param dagbag: DagBag containing DAGs with import errors
         """
+        all_parsed_files = copy.copy(dagbag.file_last_changed)
```
Comment (jedcunningham) [inline airflow/dag_processing/processor.py]: If we aren't mutating this, do we still need to `copy` it?

File: airflow/dag_processing/processor.py [hunk 4]
```diff
@@ -528,23 +529,36 @@ def update_import_errors(session: Session, dagbag: DagBag) -> None:
...
+        files_without_error = all_parsed_files - dagbag.import_errors.keys()
```
Comment (uranusjr) [inline airflow/dag_processing/processor.py]:
```suggestion
        files_without_error = dagbag.file_last_changed - dagbag.import_errors.keys()
```

**Review by potiuk — APPROVED** [review]

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**